opus/application/static_media/js/hash.js

```diff
@@ -45,6 +46,67 @@ var o_hash = {
         return hash.join("&");
     },
 
+    encodeSlugValue: function(slugValueArray) {
+        /**
+         * Take in a slug value array (like opus.selections, each element
+         * will be a value for the slug) and encode all values in the
+         * array. Return an array that contains encoded values for the
+         * slug. This function will be called in updateHash to make sure
+         * slug values in the hash are all encoded before updating the URL.
+         */
+        let slugValue = [];
+        $.each(slugValueArray, function(index, val) {
+            let value = encodeURIComponent(val);
+            value = value.replace(/\%20/g, "+");
+            slugValue.push(value);
+        });
+
+        return slugValue;
+    },
+
+    encodeHashArray: function(hashArray) {
+        /**
+         * Take in a hash array (each element will be "slug=value") and
+         * encode the "value" of each element. Return a hash array that
+         * is consist of "slug=encodedValue". This function will be called
+         * in opus.normalizedURLAPICall to make sure slug values from
+         * new URL are decoded.
+         */
+        let hash = [];
+        $.each(hashArray, function(index, pair) {
+            let slug = pair.split('=')[0];
+            let value = pair.split('=')[1];
```

Comment:
It's possible for a search string to have an "=" in it. For example, try searching on "a=b". Because the URL will say "searchslug=a=b", the split will break this into 3 parts, and we will only take "a" as the value. You can easily try this by searching a string field on "a=b" and notice that the URL will only say "a". It would be better to take value = ALL of the string after the first "=".